beacon-chain/sync/subscriber.go

```diff
@@ -198,8 +198,44 @@ func (s *Service) spawn(f func()) {
 	}
 }
 
+type subscriptionRegistrations struct {
+	sync.RWMutex
+	digests map[[4]byte]struct{}
```

Comment:
The `digests` map will only grow in size right?
If so, it's a memory leak, but I think for this specific case is totally harmless, even with a ton of BPO forks.
However, maybe it's worth adding a little comment about this "assumed memory leak".

Reply:
Note: It's exactly this kind of grow only map field of struct that used to OOM the slasher.